pydantic/_internal/_validators.py

```diff
@@ -157,7 +157,9 @@ def __call__(self, __input_value: int | float | str, **_kwargs: Any) -> Decimal:
                 raise PydanticKnownError('finite_number')
 
             if self.check_digits:
-                if exponent >= 0:
+                if isinstance(exponent, str):
+                    raise PydanticKnownError('finite_number')
+                elif exponent >= 0:
```

Comment:
mypy show  the type error of comparing `literal` and  `int`
`check_digits` feature doesn't work for Inf and NaN values.

Reply:
i think this is fine, but please add a comment explaining why we need the above.